web/src/app/schedule-view/component.ts

```diff
@@ -9,95 +9,210 @@ import {Subject, Subscription} from 'rxjs/Rx';
+    selector: 'schedule-view',
+    templateUrl: './component.html',
+    styleUrls: ['./component.scss']
 })
 
 export class ScheduleViewComponent implements OnInit, OnDestroy, AfterViewInit {
-  @ViewChildren(ScheduleComponent)
-  public ScheduleList: QueryList<ScheduleComponent>
-
-  height: number = 600;
-  private emptyScheduleSet: ScheduleSet = new ScheduleSet([], this.height);
-
-  isLoaded: boolean = false;
-  scheduleSet: ScheduleSet = this.emptyScheduleSet;
-  isTemporary: boolean = false;
-  scheduleNode;
-
-  private subscription;
-
-  constructor (
-    private selectionService : SelectionService,
-    private activatedRoute: ActivatedRoute) {
-    this.subscription = this.selectionService.subscribe(() => {
-      this.getSchedules();
-    });
-  }
-
-  public ngOnDestroy (): void {
-    this.subscription.unsubscribe();
-  }
-
-  public ngOnInit (): void {
-    this.getSchedules();
-  }
-
-  public ngAfterViewInit (): void {
-    this.ScheduleList.changes.subscribe((comps: QueryList <ScheduleComponent>) => {
-      this.scheduleNode = comps.first.scheduleNode;
-    });
-  }
-
-  private getSchedules (): void {
-    this.isLoaded = false;
-    const sectionIds = this.selectionService.getSelectedSectionIds();
-    Schedule
-      .where({ section_id: sectionIds})
-      .includes('sections')
-      .includes('sections.listing')
-      .all().then((schedules) => {
-        // TODO: Handle this in yacs-api-client
-        schedules.data.forEach(schedule => {
-          schedule.sections.forEach(section => {
-            section.periods.forEach(period => {
-              period.section = section;
+    @ViewChildren(ScheduleComponent)
+    public ScheduleList: QueryList<ScheduleComponent>
+
+    height: number = 600;
+    private emptyScheduleSet: ScheduleSet = new ScheduleSet([], this.height);
+
+    isLoaded: boolean = false;
+    scheduleSet: ScheduleSet = this.emptyScheduleSet;
+    scheduleNode;
+
+    // false if viewing temporary schedule, otherwise true
+    notTemporary: boolean = true;
+    finishedInit: boolean = false;
+    
+    private subscription;
+
+    constructor (
+        private selectionService : SelectionService,
+        private router : Router,
+        private activatedRoute: ActivatedRoute) {
+        this.subscription = this.selectionService.subscribe(() => {
+            if (this.notTemporary && this.finishedInit) this.getSchedules();
+        });
+    }
+
+    public ngOnDestroy (): void {
+        this.subscription.unsubscribe();
+    }
+
+    public ngOnInit (): void {
+        // check if schedule link entered
+
+        // maybe do this with ActivatedRouteSnapshot later
+
+        /* TODO: check if schedule link supports multiple terms.
+        Probably doesn't so will need to be implemented.
+        */
+
+        /* TODO: schedule index setting is not yet implemented */
+
+        let url: string = window.location.href;
+        let i1: number = url.indexOf('schedules?section_ids=');
```

Comment:
I would suggest refactoring this to use https://angular.io/api/router/ActivatedRoute here before merging, I think it would make this code a lot cleaner